src/verinfast/agent.py

```diff
@@ -650,13 +650,24 @@ def scanRepos(self):
             localrepos = self.config.config['local_repos']
             if localrepos:
                 for repo_path in localrepos:
+
+                    # Check for a branch
+                    split_url = repo_path.split("@")
+                    branch = None
+                    if len(split_url) == 2:
+                        branch = split_url[1]
+                        repo_path = split_url[0]
+
                     a = Path(repo_path).absolute()
                     match = re.search(r"([^/]*\.git.*)", str(a))
                     if match:
                         repo_name = match.group(1)
                     else:
                         repo_name = os.path.basename(os.path.normpath(repo_path))
-                    self.parseRepo(repo_path, repo_name)
+                    if branch is None:
```

Comment:
you actually don't need to do this check, there's an if none in that function, isn't there?